add blog post about feedback for error messages (#1566)
scala/scala-lang: blog/_posts/2023-10-17-feedback-wanted-error-messages.md

---
layout: blog-detail
post-type: blog
by: Jamie Thompson (Scala Center)

> title: "We want to know which errors bug you the most"

Reviewer: Maybe a shorter title

```suggestion
title: "Which errors bug you the most"
```

description: "Feedback is wanted on confusing or unhelpful error messages"
---

> ## We want your feedback to improve error reporting

Reviewer: ```suggestion
## Your feedback to improve error reporting
```

Author: I made it shorter

![example error message for a non-exhaustive pattern in a for-comprehension]({{ site.baseurl }}/resources/img/blog/2023-10-error-messages/splash-image.jpg)
**TL;DR: [click here][issue-template] to tell the Scala 3 core maintainers which error messages are unhelpful, confusing or otherwise suboptimal.**

### Why are we asking you

From [suggesting to import missing extension methods](https://docs.scala-lang.org/scala3/book/ca-extension-methods.html#import-extension-method) and more, Scala 3 tries to improve the ways that warnings and errors are reported to the user.

> The idea being that when the user engages with reported problems, it should feel like having a conversation with a helpful tutor. By helping users to get back on track, we can make Scala easier to learn and make Scala friendlier to newcomers, and also keep experienced users more productive.

Reviewer: ```suggestion
The idea being that when the user engages with reported problems, it should feel like having a conversation with a helpful tutor. By helping users to get back on track, we can make Scala easier to learn and friendlier to newcomers, and also keep experienced users more productive.
```

Author: thanks!

We know there is much more that can be done to continually improve the situation. As well as working to [expose diagnostics programatically](https://github.com/lampepfl/dotty/issues/14904), we are starting a new campaign to seek out confusing, annoying or unhelpful error/warning messages.

> As compiler authors, we do try to provide good messages, but its possible for there to be shortcomings.

Reviewer: ```suggestion
As compiler authors, we do try to provide good messages, but it is possible for there to be shortcomings.
```

Author: thanks!
